integration-tests/smoke/ccip/ccip_token_transfer_test.go

```diff
@@ -201,45 +191,40 @@ func TestTokenTransfer(t *testing.T) {
 					Amount: oneE18,
 				},
 			},
-			receiver:  state.Chains[sourceChain].Receiver.Address(),
-			data:      []byte("this should be reverted because gasLimit is too low, no tokens are transferred as well"),
-			extraData: changeset.MakeEVMExtraArgsV2(1, false),
-			expectedTokenBalances: map[common.Address]*big.Int{
+			Receiver:  state.Chains[sourceChain].Receiver.Address(),
+			Data:      []byte("this should be reverted because gasLimit is too low, no tokens are transferred as well"),
+			ExtraArgs: changeset.MakeEVMExtraArgsV2(1, false),
+			ExpectedTokenBalances: map[common.Address]*big.Int{
 				selfServeSrcToken.Address(): big.NewInt(0),
 				srcToken.Address():          big.NewInt(0),
 			},
-			expectedExecutionState: changeset.EXECUTION_STATE_FAILURE,
+			ExpectedStatus: changeset.EXECUTION_STATE_FAILURE,
 		},
 	}
 
-	for _, tt := range tcs {
-		t.Run(tt.name, func(t *testing.T) {
-			initialBalances := map[common.Address]*big.Int{}
-			for token := range tt.expectedTokenBalances {
-				initialBalance := changeset.GetTokenBalance(ctx, t, token, tt.receiver, e.Chains[tt.dstChain])
-				initialBalances[token] = initialBalance
-			}
+	startBlocks, expectedSeqNums, expectedExecutionStates, expectedTokenBalances :=
+		changeset.TransferMultiple(ctx, t, e, state, tcs)
 
-			changeset.TransferAndWaitForSuccess(
-				ctx,
-				t,
-				e,
-				state,
-				tt.srcChain,
-				tt.dstChain,
-				tt.tokenAmounts,
-				tt.receiver,
-				tt.data,
-				tt.expectedExecutionState,
-				tt.extraData,
-			)
+	err = changeset.ConfirmMultipleCommits(
+		t,
+		e.Chains,
+		state.Chains,
+		startBlocks,
+		false,
+		expectedSeqNums,
+	)
+	require.NoError(t, err)
 
-			for token, balance := range tt.expectedTokenBalances {
-				expected := new(big.Int).Add(initialBalances[token], balance)
-				changeset.WaitForTheTokenBalance(ctx, t, token, tt.receiver, e.Chains[tt.dstChain], expected)
-			}
-		})
-	}
+	execStates := changeset.ConfirmExecWithSeqNrsForAll(
```

Comment:
are parallel checks not applicable here like you did for commit?

Reply:
Implementation of that function is already parallel